### DISCUSSION TITLE
Translating characters to safe ones …

### DISCUSSION THREAD

**Discussion by mcepl** [languitar/pass-git-helper#381] [Q&A]
I am using pass store synchronized with my Android phone via SyncThing. Unfortunately, I don’t know if it’s limitation of Android or SyncThing, I cannot have some characters in filename. Therefore, I would need `pass-git-helper` when it would like to call `pass show dev/redcrew.org:587/mcepl@cepl.eu` called `pass show dev/redcrew.org-587/mceplAtcepl.eu` instead. Is it possible?

**Comment by languitar** [comment]
I think this is the first time I know of where the port is included in the name of the entry. So we'd need some translation table as part of the configuration for special characters? Or which solution would you prefer?

**Reply by mcepl** [reply]
That’s what `git send-email` uses with this configuration:

```
[sendemail]
›       from = Matěj Cepl <mcepl@cepl.eu>
›       smtpEncryption = tls
›       smtpServer = redcrew.org
›       smtpServerPort = 587
›       smtpUser = mcepl@cepl.eu
›       smtpAuth = "PLAIN LOGIN"
›       annotate = yes
›       assume8bitEncoding = UTF-8
```

I guess I was thinking about something in the mapping file like (just brainstorming, not sure whether it is a good design):

```ini
translate=@/At,:/-
```

or perhaps

```ini
filter=sed -e 's!@!At!g' -e 's/:/-/g'
```

which leads to running this before `pass show` is invoked:

```sh
 $ echo 'dev/redcrew.org:587/mcepl@cepl.eu'|sed -e 's!@!At!g' -e 's/:/-/g'
dev/redcrew.org-587/mceplAtcepl.eu
 $ 
```

In the second case, I am not sure how much we care about security implications of the user being able to configure running of an arbitrary command on the `pass` line (or at least `pass show` parameter).

**Reply by languitar** [reply]
Oh, I never thought about `git send-email`. But you want to continue having the port in the entry? Or should we omit that completely to get rid of the colon?

**Comment by mcepl** [comment]
On Mon Apr 29, 2024 at 4:47 PM CEST, Johannes Wienke wrote:
> Oh, I never thought about `git send-email`. But you want to continue having the port in the entry? Or should we omit that completely to get rid of the colon?

Is it something you can change? Isn't the decision on git send-email, which apparently wants that port in the key? Otherwise, I personally don't care, as long as it works with unchanged git send-email.

**Reply by languitar** [reply]
I suspect the port is part of the `host` field in the [credentials protocol](https://git-scm.com/docs/git-credential). However, I could add an option to strip a port in case it exists.

**Reply by mcepl** [reply]
Yes:

> **host**
>
> The remote hostname for a network credential. This includes the port number if one was specified (e.g., "example.com:8088").

However, I believe that the question about having port noticed is orthogonal to this discussion. If we have a system to translate some characters to some other strings, it wouldn’t matter whether we have it there or not. If we don’t, `@` still remains.

**Reply by languitar** [reply]
Implementation and mechanisms would be orthogonal. However, if both of them can solve your problem, I'd always be inclined to implement the simpler one.
